src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -82,15 +82,41 @@ public String getTenant() {
   public static Future<Token> validate(TokenValidationContext context) {
     Token token;
     try {
-      token = parse(context.getTokenToValidate(), context.getTokenCreator());
+      token = validateTokenContent(context.getTokenToValidate(), context.getTokenCreator());
     } catch (TokenValidationException e) {
       return Future.failedFuture(e);
-    } catch (Exception e) {
-      return Future.failedFuture(new TokenValidationException("Unexpected token parse exception", e, 500));
     }
 
     // Call the validateContext implementation of the underlying token type (AccessToken,
     // RefreshToken, etc.). See those classes for the validation logic specific to each type.
+    return validateTokenType(token, context);
+  }
+
+  /**
+   * Validates if token payload is correct and parsed successfully.
+   * @param tokenToValidate the token content payload to validate
+   * @param tokenCreator the token creator
+   * @return token if token content parsed successfully
+   * @throws TokenValidationException if token content could not be parsed
+   */
+  public static Token validateTokenContent(String tokenToValidate, TokenCreator tokenCreator) throws TokenValidationException {
+    try {
+      return parse(tokenToValidate, tokenCreator);
+    } catch (TokenValidationException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new TokenValidationException("Unexpected token parse exception", e, 500);
+    }
+  }
+
+  /**
+   * Validates token type, each underlying token type token can have the specific rules to check if token valid.
+   * See those classes for the validation logic specific to each type.
+   * @param token the token to validate
+   * @param context the context used for token validation
+   * @return future with token is validation was successful or failed future otherwise
+   */
+  public static Future<Token> validateTokenType(Token token, TokenValidationContext context) {
     return token.validateContext(context);
   }
```

Comment:
Not sure if this method adds any value. Why not just call `validateContext` directly?

Reply:
Because currently method validateContext is protected abstract, and I did not want to make it public. Also in case of applying this change need to do public of all 6 implementations

Reply:
I see ok. You're not wanting to call `parse` a second time. That's fine. But how about just naming your new method `validate`. So `validateTokenType(Token token, TokenValidationContext context)` becomes `validate(Token token, TokenValidationContext context)`. Because I think here it's really just another signature of validation which can contain the token for convenience in case it has already been parsed..

Reply:
Agree, committed change